.github/workflows/release.yml

```diff
@@ -3,60 +3,109 @@ name: Release
+        default: false
 
-permissions: read-all
+permissions:
+  contents: write # this is the permission that allows creating a new release
 
 # will cancel previous workflows triggered by the same event and for the same ref for PRs or same SHA otherwise
 concurrency:
-  group: ${{ github.workflow }}-${{ github.event_name }}-${{ contains(github.event_name, 'pull_request') && github.event.pull_request.head.ref || github.sha }}-${{ inputs.deploy-to }}
+  group: "${{ github.workflow }}-${{ github.event_name }}-${{ inputs.version }}-${{ inputs.deploy-to }}"
   cancel-in-progress: true
 
 jobs:
-  release:
-    name: PyPI - ${{ inputs.deploy-to }}
-    runs-on: ubuntu-latest
-    environment:
-      name: ${{ inputs.deploy-to }}
-      url: ${{ vars.PYPI_PROJECT_URL }}
-    permissions:
-      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+  release-prep:
+    name: "Release prep: generate changelog, bump version"
+    uses: dbt-labs/dbt-postgres/.github/workflows/release_prep_hatch.yml@main
+    with:
+      branch: ${{ inputs.branch }}
+      version: ${{ inputs.version }}
+      deploy-to: ${{ inputs.deploy-to }}
+    secrets: inherit
 
+  build-release:
+    name: "Build release"
+    needs: release-prep
+    runs-on: ubuntu-latest
+    outputs:
+      archive-name: ${{ steps.archive.outputs.name }}
     steps:
-      - name: Check out repository
+      - name: "Checkout ${{ github.event.repository.name }}@${{ needs.release-prep.outputs.release-branch }}"
         uses: actions/checkout@v4
         with:
+          ref: ${{ needs.release-prep.outputs.release-branch }}
           persist-credentials: false
-          ref: "${{ inputs.ref }}"
 
-      - name: Setup `hatch`
+      - name: "Setup `hatch`"
         uses: dbt-labs/dbt-adapters/.github/actions/setup-hatch@main
 
-      - name: Inputs
-        id: release-inputs
+      - name: "Set archive name"
+        id: archive
         run: |
-          version=$(hatch version)
-          archive_name=dbt-postgres-$version-${{ inputs.deploy-to }}
-          echo "archive-name=$archive_name" >> $GITHUB_OUTPUT
+          archive_name=${{ github.event.repository.name }}-${{ inputs.version }}-${{ inputs.deploy-to }}
+          echo "name=$archive_name" >> $GITHUB_OUTPUT
 
-      - name: Build `dbt-postgres`
+      - name: "Build ${{ github.event.repository.name }}"
         uses: dbt-labs/dbt-adapters/.github/actions/build-hatch@main
         with:
-          archive-name: ${{ steps.release-inputs.outputs.archive-name }}
+          archive-name: ${{ steps.archive.outputs.name }}
 
-      - name: Publish to PyPI
+  pypi-release:
+    name: "PyPI release"
+    if: ${{ !failure() && !cancelled() && !inputs.only_docker }}
+    runs-on: ubuntu-latest
+    needs: build-release
+    environment:
+      name: ${{ inputs.deploy-to }}
+      url: ${{ vars.PYPI_PROJECT_URL }}
+    permissions:
+      id-token: write  # this permission is required for trusted publishing
```

Comment:
nit: required by what?

Reply:
It's required by the PyPA action? I'm not really sure what you're asking. I did update the comment to point to the [reference](https://github.com/marketplace/actions/pypi-publish) though.